azurerm/resource_arm_network_interface.go

```diff
@@ -560,29 +585,38 @@ func expandAzureRmNetworkInterfaceIpConfigurations(d *schema.ResourceData) ([]ne
 
 		subnet_id := data["subnet_id"].(string)
 		private_ip_allocation_method := data["private_ip_address_allocation"].(string)
+		private_ip_address_version := network.IPVersion(data["private_ip_address_version"].(string))
 
 		allocationMethod := network.IPAllocationMethod(private_ip_allocation_method)
 		properties := network.InterfaceIPConfigurationPropertiesFormat{
-			Subnet: &network.Subnet{
-				ID: &subnet_id,
-			},
 			PrivateIPAllocationMethod: allocationMethod,
+			PrivateIPAddressVersion:   private_ip_address_version,
 		}
 
-		subnetId, err := parseAzureResourceID(subnet_id)
-		if err != nil {
-			return []network.InterfaceIPConfiguration{}, nil, nil, err
+		if private_ip_address_version == network.IPv4 && subnet_id == "" {
+			return nil, nil, nil, fmt.Errorf("Required field `ip_configuration`, `subnet_id` must be specified when using IPv4.")
```

Comment:
this'd be better as:

```suggestion
			return nil, nil, nil, fmt.Errorf("A Subnet ID must be specified for an IPv4 Network Interface.")
```